src/beta_inc.jl

```diff
@@ -0,0 +1,876 @@
+        x = h/(1.0 + h)
+        d = a + (b - 0.5)
+        x² = x*x
+        s₃ = 1.0 + (x + x²)
+        s₅ = 1.0 + (x + x²*s₃)
+        s₇ = 1.0 + (x + x²*s₅)
+        s₉ = 1.0 + (x + x²*s₇)
+        s₁₁ = 1.0 + (x + x²*s₉)
+    else
+        h = a/b
+        c = h/(1.0 + h)
+        x = 1.0/(1.0 + h)
+        d = b + a - 0.5
+        x² = x*x
+        s₃ = 1.0 + (x + x²)
+        s₅ = 1.0 + (x + x²*s₃)
+        s₇ = 1.0 + (x + x²*s₅)
+        s₉ = 1.0 + (x + x²*s₇)
+        s₁₁ = 1.0 + (x + x²*s₉)
+    end
+
+    # SET W = del(b) - del(a+b)
+    t = (1.0/b)^2.0
+    w = @horner(t, .833333333333333E-01, -.277777777760991E-02*s₃, .793650666825390E-03*s₅, -.595202931351870E-03*s₇, .837308034031215E-03*s₉, -.165322962780713E-02*s₁₁)
+    w *= c/b
+
+    #COMBINING 
+    u = d*log1p(a/b)
+    v = a*(log(b) - 1.0)
+    if u <= v
+        return w - u - v
+    end
+    return w - v - u
+end 
+
+"""
+    bcorr(a0,b0)
+
+Compute del(a0) + del(b0) - del(a0 + b0) where del(a) is given by:
+```math
+log(\\Gamma(a)) = (a - 0.5)*log(a) - a + 0.5*log(2\\pi) + del(a)
+```
+for a0,b0 >= 8
+"""
+function bcorr(a0::Float64, b0::Float64)
+    a = min(a0,b0)
+    b = max(a0,b0)
+
+    h = a/b
+    c = h/(1.0 + h)
+    x = 1.0/(1.0 + h)
+    x² = x*x
+    #SET SN = (1-X^N)/(1-X)
+    s₃ = 1.0 + (x + x²)
+    s₅ = 1.0 + (x + x²*s₃)
+    s₇ = 1.0 + (x + x²*s₅)
+    s₉ = 1.0 + (x + x²*s₇)
+    s₁₁ = 1.0 + (x + x²*s₉)
+    t = (1.0/b)^2.0
+    w = @horner(t, .833333333333333E-01, -.277777777760991E-02*s₃, .793650666825390E-03*s₅, -.595202931351870E-03*s₇, .837308034031215E-03*s₉, -.165322962780713E-02*s₁₁)
+    w *= c/b
+    # COMPUTE DEL(A) + W
+    t = inv(a)^2.0
+    return @horner(t, .833333333333333E-01, -.277777777760991E-02, .793650666825390E-03, -.595202931351870E-03, .837308034031215E-03, -.165322962780713E-02)/a + w
+end
+
+"""
+    esum(mu,x)
+
+Compute ``e^{mu+x}``
+"""
+function esum(mu::Float64, x::Float64)
+    if x > 0.0
+        if mu > 0.0 || mu + x < 0.0
+            w = mu
+            return exp(w)*exp(x)
+        else
+            return exp(mu + x)
+        end
+    elseif mu < 0.0 || mu + x > 0.0
+        w = mu
+        return exp(w)*exp(x)
+    else
+        return exp(mu + x)
+    end
+end
+
+"""
+    brcmp1(mu,a,b,x,y,case)
+
+If case == 1: compute ``e^{mu} * x^{a}y^{b}/B(a,b)``
+else : compute above with mu = 0.0
+"""
+function brcmp1(mu::Float64,a::Float64,b::Float64,x::Float64,y::Float64,case::Bool)
+    a0 = min(a,b)
+    b0 = max(a,b)
+    if a0 >= 8.0
+        if a > b 
+            h = b/a
+            x0 = 1.0/(1.0 + h)
+            y0 = h/(1.0 + h)
+            lambda = (a+b)*y - b
+        else
+             h = a/b
+             x0 = h/(1.0 + h)
+             y0 = 1.0/(1.0 + h)
+             lambda = a - (a+b)*x
+        end
+        e = -lambda/a
+        if abs(e) > 0.6
+            u = e - log(x/x0)
+        else
+            u = -SpecialFunctions.log1pmx(e)
+        end
+        e = lambda/b
+        if abs(e) > 0.6
```

Comment:
use a ternary here as well
```suggestion
        v = abs(e) > 0.6 ? e - log(y/y0) : -log1pmx(e)
```